app/mailers/shopping_list_mailer.rb

```diff
@@ -0,0 +1,39 @@
+class ShoppingListMailer < ApplicationMailer
+  def shopping_list_email
+    @shopping_list = params[:shopping_list]
+    email = @shopping_list.shopping_list_email
+    @file_name = "shopping_list-#{@shopping_list.name}-#{@shopping_list.shopping_day}"
+    case @shopping_list.shopping_list_email.file_format
```

Comment:
To make code more DRY, instead of calling all those methods (which have similar names) you can use `public_send` and string interpolation for dynamically preparing method name. So it would look like this:
```public_send("export_#{@shopping_list.shopping_list_email.file_format}")